modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointHistory.java

```diff
@@ -211,7 +213,7 @@ private void updateEarliestCpMap(CheckpointEntry entry) {
                     iter.remove();
             }
 
-            addCpToEarliestCpMap(entry);
+            addCpGroupStatesToEarliestCpMap(entry, entry.groupState(cctx));
```

Comment:
You can use variable 'states' here because entry.groupState(cctx) was called in this method a little earlier.

Reply:
Done.